.github/workflows/deploy-artifact.yml

```diff
@@ -83,3 +83,65 @@ jobs:
           asset_path: ${{ steps.release.outputs.artifacts_archive_path }}
           asset_name: ${{ steps.release.outputs.artifacts_archive_path }}
           asset_content_type: application/zip
+
+  deploy-to-docker:
+    runs-on: ubuntu-latest
+    needs: [test]
+
+    env:
+      OWNER: "camunda"
+      IMAGE_NAME: "zeebe-process-test-engine"
+      TAG: "latest"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@v2.4.0
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/data/common/github.com/actions/camunda-cloud/zeebe-process-test REGISTRY_HUB_DOCKER_COM_USR;
+            secret/data/common/github.com/actions/camunda-cloud/zeebe-process-test REGISTRY_HUB_DOCKER_COM_PSW;
+
+      - name: Set up Java environment
+        uses: actions/setup-java@v2.5.0
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Build jar
+        run: |
+          cd engine
+          mvn clean package -DskipTests
+
+      - name: Build Docker image
+        run: |
+          if ! echo $TAG | grep SNAPSHOT; then
+            echo ${{ github.event.release.tag_name }} >> $TAG
```

Comment:
❌ I don't think this does what you want. I think this doesn't replace the value of `TAG`. Please correct me if I'm wrong, perhaps the env-vars make this work differently than I expect. I would've expected something like `TAG="${{ github.event.release.tag_name }}"` for this to work.